sc-poc/managers/configman/adapter/file/file.go

```diff
@@ -10,29 +11,31 @@ import (
 )
 
 type File struct {
-	Path   string
-	logger *zap.Logger
+	lock          sync.RWMutex
+	Path          string
+	logger        *zap.Logger
+	configuration common.ConfigType
 }
 
 // MakeFileAdapter returns the File adapter object.
 func MakeFileAdapter(path string) adapter.Adapter {
 	logger, _ := zap.NewDevelopment()
 	file := &File{
-		Path:   path,
-		logger: logger,
+		Path:          path,
+		logger:        logger,
+		configuration: make(common.ConfigType),
 	}
 	return file
 }
 
 // GetRawConfig returns the final config.
 func (f *File) GetRawConfig() (common.ConfigType, error) {
 	// Load SC config file from file system
-	configuration, err := f.loadConfiguration()
-	if err != nil {
+	if err := f.loadConfiguration(); err != nil {
 		return nil, err
 	}
 
-	return configuration, nil
+	return f.configuration, nil
```

Comment:
Don't return `f.configuration` directly. First acquire a read lock, make a copy of f.configuration and then return it. I suggest you put the above in a function named `readConfig`. Return it directly can still result in a data race condition